### ISSUE TITLE
M-d (corfu-popupinfo-documentation) is active after completion

### ISSUE THREAD

**Issue by dr-scsi** [minad/corfu#254]
Hi,

I'm trying the new `corfu-popupinfo.el` and have the issue that after a completion, <kbd>M-d</kbd> stays active in the buffer and replaces `kill-word`.  I can place the point somewhere in the buffer and do <kbd>C-h k M-d</kbd> and get:

> `M-d` runs the command corfu-popupinfo-documentation (found in
> corfu-popupinfo-map), which is an interactive native-compiled Lisp
> function in ‘corfu-popupinfo.el’.
> 
> It is bound to `M-d`.
> 
> (corfu-popupinfo-documentation)
> 
> Show or hide documentation in popup.
> Behaves like `corfu-popupinfo-toggle`.

My config is basically this:
```
(setq corfu-popupinfo-delay nil)
(global-corfu-mode)
(corfu-popupinfo-mode)
```

Sorry but I haven't found a reliable recipe (yet?) to reproduce this, but I get the behavior after restarting and/or hitting <kbd>C-g</kbd>.  Any idea what's going wrong?

**Comment by minad** [comment]
Thanks for the report. Can you please check if corfu-popupinfo--teardown is executed correctly and in the correct buffer?

Check in particular if the modification to the key maps is undone: 

https://github.com/minad/corfu/blob/4bd40f82f112f6ad510cd3f81afe2f0db4fa2fa0/extensions/corfu-popupinfo.el#L465-L467

**Comment by dr-scsi** [comment]
@minad - Thanks.  I will try it and report back.  I have to find a way to trigger the issue again.

**Comment by ctrl2wei** [comment]
I encountered the same problem when manually entering the characters of completion option.

emacs-version: 29.0.50
emacs-repository-version: 690f7ac86ad9a9d714b1107d05c5e856a43bb18d
(pkg-info-version-info "corfu")  -> "0.31"

[Screencast from 2022-11-21 23-11-45.webm](https://user-images.githubusercontent.com/35016457/203090336-d4ac6416-80ec-402a-89ee-ef0d170808ec.webm)

**Comment by minad** [comment]
Unfortunately I am unable to reproduce the issue. In my setup, the `minor-mode-overriding-map-alist` is reset successfully after Corfu completion. Please let me know if you manage to reproduce the issue reliably. Please use Emacs 28 and start from emacs -Q, with only `global-corfu-mode` and `corfu-popupinfo-mode` enabled.

**Comment by dr-scsi** [comment]
@minad - Thanks to @ctrl2wei for the recipe, I can reproduce this with Emacs 28.2 (My first report was with Emacs 29, both on Win10).  Here is what I did:

- Cloned the relevant packages from Github under a new directory `~/lisp`
- Fetched the latest Emacs 28.2 tarball, built it with `./configure --without-native-compilation && make`.
- Started `emacs -Q &` from the `src/` directory
- Pasted and eval'ed some init forms (see below)
- Tried what ctrl2wei described and <kbd>M-d</kbd>after the completion is finished is still `corfu-popupinfo-documentation`.

It looks like this:
![corfu](https://user-images.githubusercontent.com/59777257/203149650-e473f5e7-28aa-4045-9e6b-fe7a2be27740.gif)

I've also instrumented the function `corfu-popupinfo--teardown`; it seems to work Ok in both cases.  So I suspect the issue lies somewhere else.
